From 1ade2aa21a1121e84fc89c1b4a33feac7cec49b9 Mon Sep 17 00:00:00 2001
From: btafesse <btafesse@mpr.com>
Date: Fri, 5 Apr 2019 20:09:39 -0400
Subject: [PATCH] disable HDMI

---
 arch/arm/mach-imx/video.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-imx/video.c b/arch/arm/mach-imx/video.c
index 0f94641..fec1d1f 100644
--- a/arch/arm/mach-imx/video.c
+++ b/arch/arm/mach-imx/video.c
@@ -15,6 +15,8 @@ int board_video_skip(void)
 	int ret = 0;
 	char const *panel = env_get("panel");
 
+	return -EINVAL;
+
 	if (!panel) {
 		for (i = 0; i < display_count; i++) {
 			struct display_info_t const *dev = displays+i;
@@ -76,6 +78,7 @@ int board_video_skip(void)
 #include <asm/io.h>
 int detect_hdmi(struct display_info_t const *dev)
 {
+	return -EINVAL;
 	struct hdmi_regs *hdmi	= (struct hdmi_regs *)HDMI_ARB_BASE_ADDR;
 	return readb(&hdmi->phy_stat0) & HDMI_DVI_STAT;
 }
-- 
2.7.4

