From c9b43914a4b00ad109069a6e98e191402a746f4b Mon Sep 17 00:00:00 2001
From: btafesse <btafesse@mpr.com>
Date: Fri, 5 Apr 2019 10:55:59 -0400
Subject: [PATCH] removed USBC TCPC

---
 board/freescale/common/Kconfig | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/board/freescale/common/Kconfig b/board/freescale/common/Kconfig
index 5c205bd..6d82f90 100644
--- a/board/freescale/common/Kconfig
+++ b/board/freescale/common/Kconfig
@@ -41,8 +41,3 @@ config VOL_MONITOR_LTC3882_SET
 	 This option enables LTC3882 voltage monitor set
 	 functionality. It is used by common VID driver.
 
-config USB_TCPC
-	bool "USB Typec port controller simple driver"
-	default n
-	help
-	  Enable USB type-c port controller (TCPC) driver
-- 
2.7.4

